### PR TITLE
Orlp 123.Added ability for administrator to edit, delete cards in decks.

### PR DESCRIPTION
**Pull request by IhorF**

### CODE REVIEW

File: src/app/page/admin/decks/deck/manageCards/admin.manage.cards.component.html [hunk 1]
```diff
@@ -3,29 +3,54 @@
...
+          <div class="main">
+            <textarea [disabled]="blockEdit" [(ngModel)]=card.answer>{{card.answer}}</textarea>
+          </div>
+          <button class="btn btn-primary cards-list-button" (click)="activateEdit()">Edit
+          </button>
+          <button class="btn btn-primary cards-list-button" data-toggle="modal" data-target="#deleteCard">Delete
+          </button>
+          <button *ngIf="!blockEdit" class="btn btn-primary cards-list-button" (click)="updateCard(card)">Save changes
+          </button>
+          <button *ngIf="!blockEdit" class="btn btn-primary cards-list-button" (click)="cancelEdit()">Cansel
```
Comment (antonbakalets) [inline src/app/page/admin/decks/deck/manageCards/admin.manage.cards.component.html]: Cancel?

File: src/app/page/admin/decks/deck/manageCards/admin.manage.cards.component.ts [hunk 2]
```diff
@@ -14,21 +15,23 @@ import {CreateCardDTO} from '../../../../../dto/CardsDTO/CreateCardDTO';
...
   public deck: AdminDeck;
-  public card: CreateCardDTO;
+  public card: CardPublic;
   public question: string;
   public answer: string;
   public title: string;
   public rating: '0';
   private url: string;
   private sub: Subscription;
+  public cardLink: string;
```
Comment (antonbakalets) [inline src/app/page/admin/decks/deck/manageCards/admin.manage.cards.component.ts]: I would keep _cardLink_ type as _Link_ and convert to _string_ only when used, that means in method updateCard().

Comment (IhorF) [reply src/app/page/admin/decks/deck/manageCards/admin.manage.cards.component.ts]: Ok, i will fix it.

File: src/app/page/admin/decks/deck/manageCards/admin.manage.cards.component.ts [hunk 3]
```diff
@@ -49,12 +54,50 @@ export class AdminManageCardsComponent implements OnInit {
...
+  deleteSelectedCard() {
+    this.adminManageCardsService.deleteSelectedCard(this.decodeCardLink(this.getShortCardLink(this.cardLink)))
+      .subscribe(() => {
+      this.getCardsList();
+      this.card = null;
+    });
+  }
+
+  public activateEdit() {
+    this.blockEdit = false;
```
Comment (antonbakalets) [inline src/app/page/admin/decks/deck/manageCards/admin.manage.cards.component.ts]: This is weird if you just read it.

File: src/app/dto/CardsDTO/CreateCardDTO.ts [hunk 4]
```diff
@@ -5,12 +5,14 @@ export class CreateCardDTO {
   question: string;
   answer: string;
   rating: string;
```
Comment (vldmr1703) [inline src/app/dto/CardsDTO/CreateCardDTO.ts]: Why rating is string?

File: src/app/page/admin/decks/deck/manageCards/admin.manage.cards.component.ts (hunk 3 above)
Comment (IhorF) [reply src/app/page/admin/decks/deck/manageCards/admin.manage.cards.component.ts]: Agree. I will change it.

File: src/app/dto/CardsDTO/CreateCardDTO.ts (hunk 4 above)
Comment (IhorF) [reply src/app/dto/CardsDTO/CreateCardDTO.ts]: Didn't think about it. I'll change to number.

**Pull request merged**